### PR TITLE
feat(controller-sdk-go): services support in controller-sdk-go

### PR DESCRIPTION
**Pull request by notmaxx**

### CODE REVIEW

**Review by Cryptophobia — CHANGES_REQUESTED** [review]
See conversation here for package locations: https://github.com/teamhephy/workflow-cli/pull/33#issuecomment-420754720

File: services/services.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+	var services []api.Service
+	if err := json.Unmarshal([]byte(out), &services); err != nil {
+		return []api.Service{}, err
+	}
+
+	return services, reqErr
+}
+
+// New adds a service to an app.
```
Comment (Cryptophobia) [inline services/services.go]: Can you provide a little more detail here for GoDocs? 

https://godoc.org/github.com/teamhephy/controller-sdk-go/apps#New

For example when creating a new app in the docs above, they talk about the newly created random name of the app. Here maybe write a bit more about procfileType and pathPattern that need to be passed to create new service object. Service is related to an application. You can't create a service without having an app first. It's obvious but I think good to be documented.

File: services/services.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	if reqErr != nil && !deis.IsErrAPIMismatch(reqErr) {
+		return api.Service{}, reqErr
+	}
+	defer res.Body.Close()
+
+	d := api.Service{ProcfileType: procfileType, PathPattern: pathPattern}
+	return d, reqErr
+}
+
+// Delete service from app
```
Comment (Cryptophobia) [inline services/services.go:75]: Maybe write a bit more for the docs about the possible errors when trying to delete a service.

**Review by Cryptophobia — CHANGES_REQUESTED** [review]

**Comment by notmaxx** [comment]
@Cryptophobia done, btw, do you have autobuild for this?

File: services/services.go (hunk 2 above)
Comment (Cryptophobia) [reply services/services.go:75]: Very nice, now that is good documentation! :slightly_smiling_face:

**Review by Cryptophobia — APPROVED** [review]
:100:

**Comment by Cryptophobia** [comment]
> @Cryptophobia done, btw, do you have autobuild for this?

We can integrate Travis CI builds for PR testing. Nice idea.

EDIT: Adding travis.yml for ci builds in https://github.com/teamhephy/controller-sdk-go/pull/5

**Comment by Cryptophobia** [comment]
@notmaxx , if you want to trigger travis PR build on branches now you need to merge latest from master so you get the .travis.yml file on your branch.

**Comment by notmaxx** [comment]
@Cryptophobia done

**Review by Cryptophobia — APPROVED** [review]
LGTM

**Comment by Cryptophobia** [comment]
Merged!

**Pull request merged**